<launch>
  <arg name="usb_port"                default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"           default="4000000"/>
  <arg name="namespace"               default="r_mini_driver_node"/>

  <arg name="use_moveit"              default="false"/>
  <arg name="use_joint_state"         default="false"/>
  <arg name="use_cmd_vel"             default="false"/>
  <arg name="motor_group"             default="r_mini"/>
  <arg name="remap_joint_states"      default="false"/>
  <arg name="remap_joint_trajectory" default="false"/>
  <arg name="period"                  default="0.01"/>
  <arg name="dynamixel_info"          value="$(find r_mini_driver)/config/$(arg motor_group).yaml"/>
  <arg name="launch_rviz"                        default="false"/>

  <!-- Load robot description -->
  <param name="robot_description" command="xacro '$(find r_mini_description)/urdf/arm_test.urdf.xacro'" />
  
  <!-- Start r_mini driver -->
  <node name="$(arg namespace)" pkg="r_mini_driver" type="r_mini_driver_node"
        required="true" output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <param name="use_moveit"              value="$(arg use_moveit)"/>
    <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
    <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
    <param name="period"                  value="$(arg period)"/>
    <param name="dynamixel_info"          value="$(arg dynamixel_info)"/>
  </node>


 
  <node if="$(arg launch_rviz)"
	name="rviz" pkg="rviz" type="rviz"
	args="-d $(find r_mini_description)/rviz/robot_visualization.rviz"/>

</launch>
